hazelcast/test/src/HazelcastServerFactory.h

```diff
@@ -57,14 +48,10 @@ namespace hazelcast {
 
             private:
                 std::shared_ptr<logger> logger_;
-                std::string server_address_;
```

Comment:
The initial aim for this address was that we could have started the rc on another machine, later on we seem to not use it much. Do we have such needs at all?

Reply:
This variable is always set to `g_srvFactory->get_server_address()` so I removed it. All of its uses are replaced with `remote_controller_address()`. 

Current codebase doesn't support running RC on anywhere other than localhost because its address is statically set to 127.0.0.1 [here](https://github.com/hazelcast/hazelcast-cpp-client/blob/f3991d8663eaa61ec1d38de90c1aae18a8fd2287/hazelcast/test/src/main.cpp#L51). I have extracted this value and put it in the global function `remote_controller_address()` and left a TODO there saying this value can be read from a env variable or a command line argument some day to support this feature. But even then, it would require extra refactoring. Because some tests specifically expect the member to be on 127.0.0.1. (see [this](https://github.com/hazelcast/hazelcast-cpp-client/blob/f3991d8663eaa61ec1d38de90c1aae18a8fd2287/hazelcast/test/src/HazelcastTests1.cpp#L160) for example, it doesn't do any networking config, so the default 127.0.0.1 is used. If the RC is on a different host than tests using this function will fail.)